hack/docker-integration-test.sh

```diff
@@ -14,18 +13,78 @@ assert_success() {
 	fi
 }
 
+get_volume_id() {
+	VOLUME_ID=$(curl -X POST "http://localhost:9116/v1/volumes/filters" \
+		-H "accept: application/json" -H "Content-Type:application/json" \
+		-d "{ \"locator\": { \"name\": \"$1\"}}" \
+		| jq .[] | jq .[0] -r)
+}
+
+assert_attached(){
+	ATTACH_PATH=$(curl -X GET "http://localhost:9116/v1/volumes/inspect/$VOLUME_ID" \
+		-H "accept:application/json" | jq  ."volume" | jq ."attach_path" | jq .[0] -r)
+
+	# ATTACH_PATH is null when the volume unmounted. If it is non-null,
+	# we know that the volume is mounted. 
+	#
+	# Here we can assert that the volume is mounted
+	if [ $1 == "true" ];
+	then
+		echo "Asserting volume attached: $ATTACH_PATH"	
+		if [ $ATTACH_PATH != "null" ];
+		then
+			return 0
+		else
+			cleanup
+			exit 1
+		fi
+	# Here we can assert that the volume is unmounted
+	else
+		echo "Asserting volume detached: $ATTACH_PATH"	
+		if [ $ATTACH_PATH == "null" ];
+		then
+			return 0
+		else
+			cleanup
+			exit 1
+		fi
+	fi
+}
+
+
 # Start OSD
-make install
-sudo $GOPATH/bin/osd -d --driver=name=nfs,server=127.0.0.1,path=/nfs --sdkport 9106 --sdkrestport 9116 &
+sudo $GOPATH/bin/osd -d --driver=name=fake --sdkport 9106 --sdkrestport 9116 &
```

Comment:
Pass in here the shared secret using the environment variable (which is located in cmd/osd/main.go)